Handle width and height placeholders in the thumbnail URL

Makrit opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change

The change updates the stream_picture logic to remove the {width}x{height} placeholder from the thumbnail_url. The current thumbnail_url returned by the Twitch API includes the {width}x{height} placeholder, which needs to be set with specific dimensions for the URL to work properly. By removing the placeholder, the URL defaults to the best available resolution for the thumbnail image.

By removing the placeholder, we make sure the thumbnail image can be used directly, regardless of the resolution or 
dimensions specified.



## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #132959
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [x] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

Makrit committed:
Update coordinator.py

Updated the stream_picture logic to handle the width and height placeholders in the thumbnail URL, using the default resolution.

Makrit committed:
Merge branch 'home-assistant:dev' into dev

Makrit committed:
Merge branch 'dev' into dev

joostlek reviewed (changes requested):
Please add a rationale to this change and elaborate on what this fixes

Makrit commented:
Maybe be better to limit the resolution for bandwidth
`stream.thumbnail_url.replace("{width}x{height}", "640x360") if stream else None,`
I didn't want to make that choice.

Makrit committed:
Update coordinator.py

Added the user's channel by adding it to the list of users being processed. Ensuring follow status is always set to True and Skipped the subscription check for the logged-in user's channel

Makrit committed:
Merge branch 'home-assistant:dev' into dev

Makrit committed:
Added user sensor

Added the user's channel by adding it to the list of users being processed. Ensuring follow status is always set to True and Skipped the subscription check for the logged-in user's channel

Makrit committed:
Merge branch 'dev' of https://github.com/Makrit/core into dev

Makrit committed:
Fix displaying live stream thumbnail

Update the stream_picture logic to remove the {width}x{height} placeholder from the thumbnail_url. The current thumbnail_url returned by the Twitch API includes the {width}x{height} placeholder

Makrit commented:
Sorry for the clutter. I didn't realise further changes would go towards this one. I've changed it all back.
